### DISCUSSION TITLE
Who's using Gatus?

### DISCUSSION THREAD

**Discussion by TwiN** [TwiN/gatus#440] [Show and tell]
If you're using Gatus, I would really appreciate if you could let me know in the comments.

Regardless of whether you're using it for personal projects or at your company, a quick comment of what features you use would greatly benefit me _(DNS? TCP? HTTP? If you're using HTTP, are you just checking the status, or are you also validating the response body?)_

For instance:
- What company do you work for?
- What do you monitor with Gatus? (business applications? databases? DNS queries? certificate expiration?)
- How long have you been using it for?
- How many endpoints are you monitoring?
- Is it public-facing, or internal only? (e.g. can customers see it, or only developers?)
- What feature would you like to see in Gatus?
- What do you dislike the most about Gatus?

### Template
Though you don't have to use this template, here's a ready-to-use template with the aforementioned questions, if that helps convince you to spend a couple minutes to provide some feedback.
```markdown
**What company do you work for?:** 
**What do you monitor with Gatus?:**
**How long have you been using it for?:**
**How many endpoints are you monitoring?:**
**Is it public-facing, or internal only?:**
**What feature would you like to see in Gatus?:**
**What do you dislike the most about Gatus?:**
```

---

If you'd rather share feedback privately, please feel free to reach out to feedback@gatus.io

**Comment by zeylos** [comment]
I do use gatus in production :upside_down_face: 

**What company do you work for?:** 
Eco CO2 https://www.ecoco2.com/

**What do you monitor with Gatus?:**
Webapps frontpages, api endpoints, internal tools, static websites, certificates expiration (I use wildcards certificates so I configure specific endpoints for each certificate only to check expiration once a day).
I only use HTTP checks, with status and response body for some api endpoints.

**How long have you been using it for?:**
6 months

**How many endpoints are you monitoring?:**
approx. 50

**Is it public-facing, or internal only?:**
Internal only

**Likes/dislikes:**
  - I really like the simplicity of the tool as is, I only use it to get alerts for our websites and apis from outside of the clusters.
  - I currently have to deploy 3 gatus instances because I need to alert on 3 different teams channels, which is not optimal but actually not that problematic cause it makes the configuration smaller for each instance and easier to read.
  - I guess the dashboards could be improved in terms of UI/UX, but I've no skill in this so I can't really help :D. For my use the current UI is totally enough because I mainly use alerts.
  - For really high workloads maybe we can imagine an integration with kubernetes, where gatus would deploy pods to parallelize the checks or something ? I don't think I'll have the use of it though, but the idea went through my head so I'm writing it down.

Again, thanks for the really good job on this tool, I really appreciate working with it (and it saved our asses more than once :upside_down_face: ).
I'll populate my list above if I think of something else :)

**Comment by rissson** [comment]
I use Gatus at work and personnaly, but with approximately the same features.

**What company do you work for?:** 

- [EPITA](https://epita.fr), for the [CRI](https://blog.cri.epita.fr)
- I also use Gatus personnaly

**What do you monitor with Gatus?:**

Any user-facing services. The goal is to be notified when something doesn't work, as a last resort external monitoring tool.

**How long have you been using it for?:**

A bit more than a year!

**How many endpoints are you monitoring?:**

41 at work (a few are still missing).
12 personnaly (planning to add more).

**Is it public-facing, or internal only?:**

Work: https://devou.ps, configuration: https://gitlab.com/lama-corp/infra/dotshabka/-/blob/master/k3s/apps/gatus/overlays/devoups/config/config.yaml
Personal: https://status.lama-corp.space, configuration: https://gitlab.kektus.xyz/kektus/infra/infrastructure/-/blob/master/k8s/apps/manifests/gatus/overlays/lama-corp/config/config.yaml

**What feature would you like to see in Gatus?:**

- Being able to configure the retention time, but I believe that's planned already.
- Dark theme by default!
- The items should be sorted one the front page according to the configuration file. One way of doing this could be to add a `groups` list with the name of the group and its priority.

**What do you dislike the most about Gatus?:**

Configuration can be a chore to write, and can get quite big, but we don't change it every day, so that's alright.

**Comment by BluemediaDev** [comment]
I'm using Gatus "productively" to monitor my private services. In my case it is mainly black box monitoring, as I use it in addition to a more extensive monitoring system (Checkmk + Grafana Loki).

**What do you monitor with Gatus?:**

Mainly Webservices / Webapps, my mail server, DNS servers, an TURN server and some ICMP checks.

**How long have you been using it for?:**

In fact, only since a few days, as I was looking for a better alternative to Uptime Kuma. And I'm really glad I found Gatus.

**How many endpoints are you monitoring?:**

I'm currently monitoring 38 endpoints using Gatus.

**Is it public-facing, or internal only?:**

My instance is semi-public. Since I use Gatus mainly for my status page, I developed my own frontend for it ([Fancy Gatus](https://github.com/BluemediaGER/fancy-gatus)). It is inspired by the status page of Uptime Kuma and designed for simplicity. The live version is available here: https://status.bluemedia.dev

**What feature would you like to see in Gatus?:**

It would be nice if the average values for uptime and response time were not only available as badges, but also included in the API responses.

**What do you dislike the most about Gatus?:**

I actually really like Gatus so far. The config gets a bit clunky with many endpoints, but that's probably just due to the YAML syntax.

I would like to thank you for the great tool and will look around to see if I can help with the implementation of some features in the future.

**Comment by mdraijer** [comment]
**What company do you work for?:** 

A department of the Dutch ministry of internal affairs.

**What do you monitor with Gatus?:**

We're using Gatus for monitoring of applications that run on our platform behind a VPN. For monitoring from 'outside' we use tools in the cloud.

**How long have you been using it for?:**

About a year.

**How many endpoints are you monitoring?:**

20-30, mostly just checking on status.

**Is it public-facing, or internal only?:**

Internal only.

**What feature would you like to see in Gatus?:**

* Longer retention
* API to get the uptime in JSON format (instead of an svg xml that is harder to parse).

**What do you dislike the most about Gatus?:**

\-

**What do you like the most about Gatus?:**

That it is small and simple. Try to keep it like that!

**Comment by LouisMT** [comment]
**What company do you work for?:**
Personal projects

**What do you monitor with Gatus?:**
I manage a few home servers, mainly for home automation, network controllers and personal projects

**How long have you been using it for?:**
A few months

**How many endpoints are you monitoring?:**
34 (mostly HTTPS, one DNS, and I'm also checking the response body for some!)

**Is it public-facing, or internal only?:**
Both

**What feature would you like to see in Gatus?:**
I've got everything I need! I have been using Netdata for a while, which is great but way too advanced for my use case, resulting in a lot of spam messages! I just need to know if my most critical services are still up, which Gatus is doing a great job at, without too much configuration or resources.

**What do you dislike the most about Gatus?:**
Nothing. It's awesome.

**Comment by boopzz** [comment]
What company do you work for?
UK based public sector

What do you monitor with Gatus? (business applications? databases? DNS queries? certificate expiration?)
All HTTPS monitors to monitor tooling

How long have you been using it for?
Nearly 3 months

How many endpoints are you monitoring?
11

Is it public-facing, or internal only? (e.g. can customers see it, or only developers?)
Internal

What feature would you like to see in Gatus?

- Some easier branding or at least theme selections

What do you dislike the most about Gatus?

- When a config change is made and a site is shown as down, once the number of failed monitors is reached it re-alerts. Would be nice to be able to have a state of each service and not alert upon config change

**Comment by Doomsdayrs** [comment]
**What company do you work for?:**  ShosetsuOrg
**What do you monitor with Gatus?:** Shosetsu various endpoints, such as CDN and error reporting
**How long have you been using it for?:** Today
**How many endpoints are you monitoring?:** cdn, repo, acra, @
**Is it public-facing, or internal only?:** public
**What feature would you like to see in Gatus?:** Stats and persistent logs of website history
**What do you dislike the most about Gatus?:** A way to customize brand colors?

**Comment by j0nathontayl0r** [comment]
**What company do you work for?:** Dingo Software
**What do you monitor with Gatus?:** API, endpoint and certificate monitoring
**How long have you been using it for?:** 4 weeks against our development environment, since today for production.
**How many endpoints are you monitoring?:** 42x endpoints
**Is it public-facing, or internal only?:** Public
**What feature would you like to see in Gatus?:** Offer this as a SaaS so I don't need to wrangle CloudFormation scripts to do it for me, we'll pay for it. ;) 
**What do you dislike the most about Gatus?:** None at this time.

**Reply by TwiN** [reply]
> **What feature would you like to see in Gatus?:** Offer this as a SaaS so I don't need to wrangle CloudFormation scripts to do it for me, we'll pay for it. ;)

@j0nathontayl0r Actually, Gatus _is_ offered as a SaaS too ([gatus.io](https://gatus.io)). 

I just make it a point to not mention it on this open source project outside of the repository's about section (and [the cheeky link on the banner's README that I added 2 weeks ago](https://github.com/TwiN/gatus/commit/9d14e3011bcc5cd871fc5f8a8e383c9713d3245c)), because I feel that it goes against the nature of what open source should be 😅

**Reply by j0nathontayl0r** [reply]
Oh that's awesome! Thanks for the reply @TwiN. I thought you offered it as SaaS but I wasn't sure if I was misremembering or thinking it might have been a different monitoring tool some where else. 

Well, I'm sold. You're a damn site cheaper than hosting it myself on AWS ECS. I'll keep running it our selves for now to prove it to the business and get in touch.

**Comment by jangrewe** [comment]
**What company do you work for?:** private server
**What do you monitor with Gatus?:** All services and websites
**How long have you been using it for?:** 2 days
**How many endpoints are you monitoring?:** 192
**Is it public-facing, or internal only?:** internal only, until there's a way to hide specific endpoints for unauthorized users
**What feature would you like to see in Gatus?:** 
* hiding  specific endpoints for unauthorized users
* show/hide healthy/unhealthy endpoints toggle
* tags for endpoints (groups are nice, but too rigid)
* filter by name/tags

**What do you dislike the most about Gatus?:** no parallelization of checks (`disable-monitoring-lock: true` doesn't seem to have any effect), white blocks when scrolling, 



Yes, i may be overtaxing Gatus a bit with nearly 200 endpoints, but it's great for a high-level overview. For low-level metrics, i use Prometheus/Alertmanager.

**Reply by TwiN** [reply]
setting `disable-monitoring-lock` to `true` should definitely allow concurrent checks, but regardless of whether you have it enabled or not, there's a 777ms delay between the start of each goroutine: https://github.com/TwiN/gatus/blob/3c246f0c69243cc26933fcae750a722123bcad4d/watchdog/watchdog.go#L33

`disable-monitoring-lock` allows several endpoints to be checked at the same time thereafter https://github.com/TwiN/gatus/blob/3c246f0c69243cc26933fcae750a722123bcad4d/watchdog/watchdog.go#L55-L61

**Comment by ka-mp-6765** [comment]
I work for an MSP but are using Gatus at home for the last 3 weeks. The api let me build [the dashboard I always](https://codeberg.org/tonoit/gatus-tools) wanted (just a page with health badges, and one click to the underlying gatus page) in minutes. It's work in progress but... it's there.

It's internal use only and by now it's likely to replace my nagios core :(
There are roughly 30 devices in it (routers, switches, wifi, iot, linux) at this moment and with an average 5 tests per device but growing.

For the future I hope it won't become to bloated (especially no AI) but SNMP checks (eg monitoring ink level, interface traffic & errors on routers & switches) would be the tipping point to drop nagios core. I used a simple script for initial discovery and adding of devices but don't think that should be built in, but could very well be supplied an separate external helper.
